app/nexus.go

```diff
@@ -0,0 +1,545 @@
+	namespaceIDs []string,
+	resourceVersion string,
+	asyncOperationId string,
+) (*operation.AsyncOperation, error) {
+	existingPolicySpecs := existingEndpoint.Spec.PolicySpecs
+	namespaceIDMap := make(map[string]struct{}, len(namespaceIDs))
+	for _, namespaceID := range namespaceIDs {
+		namespaceIDMap[namespaceID] = struct{}{}
+	}
+
+	updatedPolicySpecs := make([]*nexus.EndpointPolicySpec, 0)
+	for _, existingPolicySpec := range existingPolicySpecs {
+		if _, ok := namespaceIDMap[existingPolicySpec.AllowedCloudNamespacePolicySpec.NamespaceId]; !ok {
+			updatedPolicySpecs = append(updatedPolicySpecs, existingPolicySpec)
+		}
+	}
+
+	if len(updatedPolicySpecs) == len(existingPolicySpecs) {
+		return nil, fmt.Errorf("no updates to be made")
+	}
+	existingEndpoint.Spec.PolicySpecs = updatedPolicySpecs
+	return c.callUpdateEndpoint(existingEndpoint, resourceVersion, asyncOperationId)
+}
+
+func (c *NexusClient) deleteEndpoint(
+	endpointID string,
+	resourceVersion string,
+	asyncOperationId string,
+) (*operation.AsyncOperation, error) {
+	resp, err := c.client.DeleteNexusEndpoint(c.ctx, &cloudservice.DeleteNexusEndpointRequest{
+		EndpointId:       endpointID,
+		ResourceVersion:  resourceVersion,
+		AsyncOperationId: asyncOperationId,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return resp.AsyncOperation, nil
+}
+
+func NewNexusCommand(getNexusClientFn GetNexusClientFn) (CommandOut, error) {
+	var c *NexusClient
+	endpointNameFlag := &cli.StringFlag{
+		Name:     "name",
+		Aliases:  []string{"n"},
+		Usage:    "Endpoint name",
+		Required: true,
+	}
+	targetNamespaceFlag := &cli.StringFlag{
+		Name:     "target-namespace",
+		Aliases:  []string{"tns"},
+		Usage:    "Namespace in which a handler worker will be polling for Nexus tasks on",
+		Required: true,
+	}
+	targetNamespaceFlagOptional := &cli.StringFlag{
+		Name:     targetNamespaceFlag.Name,
+		Aliases:  targetNamespaceFlag.Aliases,
+		Usage:    targetNamespaceFlag.Usage + " (optional)",
+		Required: false,
+	}
+	targetTaskQueueFlag := &cli.StringFlag{
+		Name:     "target-task-queue",
+		Aliases:  []string{"ttq"},
+		Usage:    "Task Queue in which a handler worker will be polling for Nexus tasks on",
+		Required: true,
+	}
+	targetTaskQueueFlagOptional := &cli.StringFlag{
+		Name:     targetTaskQueueFlag.Name,
+		Aliases:  targetTaskQueueFlag.Aliases,
+		Usage:    targetTaskQueueFlag.Usage + " (optional)",
+		Required: false,
+	}
+	allowNamespaceFlag := &cli.StringSliceFlag{
+		Name:     "allow-namespace",
+		Aliases:  []string{"ans"},
+		Usage:    "Namespace that is allowed to call this endpoint",
+		Required: true,
+	}
+	namespaceFlag := &cli.StringSliceFlag{
+		Name:     "namespace",
+		Aliases:  []string{"ns"},
+		Usage:    "Namespace that is allowed to call this endpoint",
+		Required: true,
+	}
+	return CommandOut{
+		Command: &cli.Command{
+			Name:    "nexus",
+			Aliases: []string{"nxs"},
+			Usage:   "Commands for managing Nexus operations (EXPERIMENTAL)",
```

Comment:
[Nit] 'operations' seems a bit weird here, but I don't have another suggestion

Reply:
I can remove the usage statement until we have commands other than the `endpoint` under `nexus`?

Reply:
makes sense